### DISCUSSION TITLE
Show existing data in defaultValue multi

### DISCUSSION THREAD

**Discussion by yaffalhakim1** [JedWatson/react-select#5566] [General]
My apologize because im new with react-select

so i already get data with GetServersideprops (nextjs) like this 
![image](https://user-images.githubusercontent.com/63090705/221347914-f87e82e8-3ee3-49a2-8dc5-6e4d795cfc14.png)

i want to use the tag as default value, so the user don't have to input it again. i do it like this 
![image](https://user-images.githubusercontent.com/63090705/221348007-6b4b95ad-aa32-44d1-8a49-efeb4f868444.png)

the problem is, the data is not showing 
![image](https://user-images.githubusercontent.com/63090705/221347999-c563c151-cd90-4c0b-aaf3-db19d8432dec.png)
it is supposed to be like this 
![image](https://user-images.githubusercontent.com/63090705/221348051-23fad8fb-7762-4560-841b-3e1ec8459917.png)

anyone have a solution? thank you.

**Comment by manjushsh** [comment]
HI @yaffalhakim1, Can you share sample response or options array or share this example on [Code Sandbox](https://codesandbox.io/)?

**Reply by yaffalhakim1** [reply]
i already solved it, the sample response is like this ["2021". "2022"]
so i change the response to this 
`data.tag.map((tag: any) => ({ value: tag, label: tag }));`

**Comment by Rall3n** [comment]
Hello @yaffalhakim1,

`react-select` does not support simple value options (strings instead of options) out of the box. We are working on improving support in this regard.

One of the hurdles for simple string support is the ability to define option groups. The `Select` component currently checks internally for an `options` attribute on an option to determine if it is a group or not using the [`... in ...`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in) operator. This would cause errors using string options as `in` is only supported in combination with objects.

Until we have a proper solution which allows more freedom in defining groups (and therefore improves string option support), your best bet is to convert the string options into objects.
You do not have to duplicate the string to provide both `label` and `value`. You can convert them into objects with a single attribute, and define `getOptionLabel` and/or `getOptionValue` to provide this attribute as label and value to the `Select` component.

**Reply by yaffalhakim1** [reply]
thank you for responding my question, i already solved it.
so my response api is like this 
["2021","2022"] 
so i convert the this value by doing it like this like your suggestion (convert to object)

`data.tag.map((tag: any) => ({ value: tag, label: tag }));`

and provide it in a defaultValue

                      `<Select
                       name="tag[]"
                      isMulti
                      options={options}
                      className="basic-multi-select"
                      onChange={handleSelectChange}
                      defaultValue={defaultValue}
                      classNamePrefix="select"
                    />`

and provide the handler
`const handleSelectChange = (selectedOptions: any) => {
    const options = selectedOptions.map((option: any) => option.label);
    setField({
      ...field,
      tag: options,
    });
  };`
